### PR TITLE
Remove old editor files and UI elements

### PR DESCRIPTION
**Pull request by SYBIOTE**
This pull request removes old editor files, updates the editor name and route, and removes various panels, dialogs, layouts, and UI elements that are no longer needed. These changes help to slim down the editor packages and remove unused components.

### CODE REVIEW

**Comment by HexaField** [comment]
Nice work, still failing errors though

**Comment by SYBIOTE** [comment]
Ill fix it up

**Comment by CITIZENDOT** [comment]
These packages can be removed from `packages/editor/package.json`:
- react-hot-keys
- rc-slider

**Comment by hanzlamateen** [comment]
Also, I think following files are redundant:

1. packages/client-core/src/admin/common/Location/LocationDrawer.tsx (It was previously used to publish locations from studio)
2. packages/client-core/src/admin/common/Project/ProjectDrawer.tsx (I do see it being used in studio's project page, but I wonder is we can use the dialog used in admin panel to get rid of this old UI drawer based control). If we get rid of this then following can be removed as well
  a. packages/client-core/src/admin/common/DrawerView.tsx
  b. packages/client-core/src/admin/common/validation/formValidation.ts

**Comment by aditya-mitra** [comment]
adding a few code comments for files which have moved or have changed lines. this will help us better understand what happened to the older lines. (ideally, remove old editor files means deletions of files)

File: packages/editor/src/components/assets/useUpload.tsx [hunk 1]
```diff
@@ -32,9 +32,6 @@ import { getState } from '@etherealengine/hyperflux'
 
 import { getEntries, uploadProjectAssetsFromUpload } from '../../functions/assetFunctions'
 import { EditorState } from '../../services/EditorServices'
-import { DialogState } from '../dialogs/DialogState'
-import ErrorDialog from '../dialogs/ErrorDialog'
-import { ProgressDialog } from '../dialogs/ProgressDialog'
```
Comment (aditya-mitra) [inline packages/editor/src/components/assets/useUpload.tsx:37]: All of these dialogs were non-functional in the new UI. Also, a full refactor of the useUpload function is needed.

File: packages/client/src/pages/editor/index.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ import { RouterState } from '@etherealengine/client-core/src/common/services/Rou
 import { PopupMenuInline } from '@etherealengine/client-core/src/user/components/UserMenu/PopupMenuInline'
 import { AuthState } from '@etherealengine/client-core/src/user/services/AuthService'
 import { userHasAccess } from '@etherealengine/client-core/src/user/userHasAccess'
-import { EditorPage, useStudioEditor } from '@etherealengine/editor/src/pages/Editor2Page'
+import { EditorPage, useStudioEditor } from '@etherealengine/editor/src/pages/EditorPage'
```
Comment (aditya-mitra) [inline packages/client/src/pages/editor/index.tsx:35]: renamed tailwind `Editor2Page` to `EditorPage`

File: packages/editor/src/components/EditorContainer.css [hunk 3]
```diff
@@ -1,76 +1,64 @@
-/* this is basically style for the editor page*/
-/* we can drill down furthur if needed*/
+/** references https://github.com/ticlo/rc-dock/blob/master/dist/rc-dock.css */
```
Comment (aditya-mitra) [inline packages/editor/src/components/EditorContainer.css:1]: renamed `Editor2Container.css` as `EditorContainer.css` and moved the contents from `Editor2Container.css` to here

File: packages/editor/src/components/EditorContainer.tsx
Comment (aditya-mitra) [inline packages/editor/src/components/EditorContainer.tsx:1]: same for this. renamed. moved contents from `Editor2Container.tsx` to `EditorContainer.tsx`

File: packages/editor/src/components/dialogs/ImportSettingsPanelDialog.tsx
Comment (aditya-mitra) [inline packages/editor/src/components/dialogs/ImportSettingsPanelDialog.tsx:1]: renaming

File: packages/editor/src/components/dialogs/SaveSceneDialog.tsx
Comment (aditya-mitra) [inline packages/editor/src/components/dialogs/SaveSceneDialog.tsx:1]: removed `SaveSceneDialog2.tsx` to `SaveSceneDialog` and moved contents from `SaveSceneDialog2.tsx` to here

File: packages/editor/src/components/toolbar/Toolbar.tsx [hunk 4]
```diff
@@ -45,8 +45,8 @@ import { onNewScene } from '../../functions/sceneFunctions'
 import { cmdOrCtrlString } from '../../functions/utils'
 import { EditorState } from '../../services/EditorServices'
 import CreateSceneDialog from '../dialogs/CreateScenePanelDialog'
-import ImportSettingsPanel from '../dialogs/ImportSettingsPanelDialog2'
-import { SaveNewSceneDialog, SaveSceneDialog } from '../dialogs/SaveSceneDialog2'
+import ImportSettingsPanel from '../dialogs/ImportSettingsPanelDialog'
+import { SaveNewSceneDialog, SaveSceneDialog } from '../dialogs/SaveSceneDialog'
```
Comment (aditya-mitra) [inline packages/editor/src/components/toolbar/Toolbar.tsx:49]: import changes due to file movement

File: packages/editor/src/components/visualScript/VisualScriptUIModule.ts [hunk 5]
```diff
@@ -23,19 +23,6 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
...
-export * from './components/InputSocket'
-export * from './components/Node'
-export * from './components/NodeContainer'
-export * from './components/NodePicker'
-export * from './components/OutputSocket'
-export * from './components/modals/ClearModal'
-export * from './components/modals/HelpModal'
-export * from './components/modals/LoadModal'
-export * from './components/modals/Modal'
-export * from './components/modals/SaveModal'
```
Comment (aditya-mitra) [inline packages/editor/src/components/visualScript/VisualScriptUIModule.ts:38]: deleted components

File: packages/editor/src/pages/EditorPage.tsx
Comment (aditya-mitra) [inline packages/editor/src/pages/EditorPage.tsx:1]: `Editor2Page` renamed as `EditorPage` and contents of `Editor2Page` moved here

File: packages/editor/src/systems/ModelHandlingSystem.ts [hunk 6]
```diff
@@ -30,8 +30,8 @@ import { PresentationSystemGroup } from '@etherealengine/ecs/src/SystemGroups'
 import BufferHandlerExtension from '@etherealengine/engine/src/assets/exporters/gltf/extensions/BufferHandlerExtension'
 import { defineActionQueue, getState } from '@etherealengine/hyperflux'
 
-import { ImportSettingsState } from '../components/assets/ImportSettingsPanel'
 import { clearModelResources, uploadProjectFiles } from '../functions/assetFunctions'
+import { ImportSettingsState } from '../services/ImportSettingsState'
```
Comment (aditya-mitra) [inline packages/editor/src/systems/ModelHandlingSystem.ts:34]: import changes

File: packages/editor/src/systems/UploadRequestSystem.tsx [hunk 7]
```diff
@@ -30,9 +30,9 @@ import { PresentationSystemGroup } from '@etherealengine/ecs/src/SystemGroups'
 import { UploadRequestState } from '@etherealengine/engine/src/assets/state/UploadRequestState'
 import { getMutableState, getState, NO_PROXY, useState } from '@etherealengine/hyperflux'
 
-import { ImportSettingsState } from '../components/assets/ImportSettingsPanel'
 import { uploadProjectFiles } from '../functions/assetFunctions'
 import { EditorState } from '../services/EditorServices'
+import { ImportSettingsState } from '../services/ImportSettingsState'
```
Comment (aditya-mitra) [inline packages/editor/src/systems/UploadRequestSystem.tsx:35]: import changes

File: packages/projects/default-project/xrengine.config.ts
Comment (aditya-mitra) [inline packages/projects/default-project/xrengine.config.ts:1]: remove old studio and studio now imports from renamed pages/editor

File: packages/editor/src/services/ImportSettingsState.ts
Comment (aditya-mitra) [inline packages/editor/src/services/ImportSettingsState.ts:1]: this file is just written with `ImportSettingsState` which was earlier present in `ImportSettingsPanel`

File: packages/projects/default-project/xrengine.config.ts
Comment (aditya-mitra) [inline packages/projects/default-project/xrengine.config.ts:1]: removed old studio path

File: packages/editor/src/services/ImportSettingsState.ts
Comment (aditya-mitra) [inline packages/editor/src/services/ImportSettingsState.ts:1]: this file is just written with ImportSettingsState which was earlier present in ImportSettingsPanel

**Review by dinomut1 — DISMISSED** [review]
This is gonna break several projects. We need to add linked PRs for
https://github.com/theinfinitereality/irpro-wizard.git
https://github.com/theinfinitereality/eepro-ecommerce-tools.git
https://github.com/theinfinitereality/eepro-xrui-components.git
before merging them all in together

**Comment by aditya-mitra** [comment]
yes here they are:
- https://github.com/theinfinitereality/eepro-ecommerce-tools/pull/97
- https://github.com/theinfinitereality/eepro-multitenancy/pull/360
- https://github.com/theinfinitereality/irpro-wizard/pull/6
- https://github.com/theinfinitereality/eepro-xrui-components/pull/24

**Comment by aditya-mitra** [comment]
> packages/client-core/src/admin/common/Project/ProjectDrawer.tsx (I do see it being used in studio's project page, but I wonder is we can use the dialog used in admin panel to get rid of this old UI drawer based control). If we get rid of this then following can be removed as well
a. packages/client-core/src/admin/common/DrawerView.tsx

only these 2 will be removed later once the projects page is refactored into tailwind (currently just used in single tenant dev)

**Comment by dinomut1** [comment]
is the ecommerce branch ready for merge yet? once that's ready i think we're good to go

**Review by HexaField — CHANGES_REQUESTED** [review]
Screenshot from 2024-07-25 13-16-56

something doesnt look right. maybe its using the wrong themeing? or from merging dev in?

**Comment by aditya-mitra** [comment]
> is the ecommerce branch ready for merge yet? once that's ready i think we're good to go

yes it is

**Comment by CITIZENDOT** [comment]
> something doesnt look right. maybe its using the wrong themeing? or from merging dev in?

Just checked this on a fresh install and I don't see this glitch. [Something similar happened to me before](https://github.com/EtherealEngine/etherealengine/pull/10542#issuecomment-2220276539), but it went away after cleaning node_modules and reinstalling them.

**Review by HexaField — APPROVED** [review]
Amazing work.

**Pull request merged**